websites/F/Free Web Novel/presence.ts

```diff
@@ -0,0 +1,192 @@
+		presence.getSetting<boolean>("logoOnly")
+	]);
+	const [privacy, showTimestamp, showReading] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("showTimestamp"),
+			presence.getSetting<boolean>("showReading")
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: "nocover"
+		},
+		{ pathname } = window.location,
+		pathnames = [
+			"/",
+			"/history/",
+			"/latest-release-novel/",
+			"/latest-novel/",
+			"/completed-novel/",
+			"/most-popular-novel/",
+			"/search/",
+			"/contact.html",
+			"/sitemap.xml",
+			"/privacy-terms-of-use.html",
+			"/genre/",
+			"/author/"
+		];
+	if (!showReading) {
+		showBook = false;
+		showCover = false;
+		showLogo = true;
+		showButtons = false;
+	} else {
+		presenceData.smallImageKey = "closed";
+		presenceData.smallImageText = "Not Reading";
+	}
+	if (showTimestamp) presenceData.startTimestamp = browsingTimestamp;
+	if (!showBook) showCover = false;
+	switch (true) {
+		case privacy: {
+			presenceData.largeImageKey = "incognito";
+
+			if (
+				(pathnames.includes(pathname) ||
+					document.querySelector<HTMLDivElement>('[class="m-imgtxt"]')) &&
+				showReading
+			)
+				presenceData.details = "Browsing...";
+			else if (
+				document.querySelector<HTMLAnchorElement>(
+					'[title="Read Next chapter"]'
+				) &&
+				showReading
+			) {
+				presenceData.details = "Reading...";
+				presenceData.smallImageKey = "open";
+			}
+			break;
+		}
+		case pathname === "/": {
+			presenceData.details = "Browsing the homepage";
+			break;
+		}
+		case pathname.includes("/history/"): {
+			presenceData.details = "Browsing recently viewed novels";
+			break;
+		}
+		case pathname.includes("/latest-release-novel/"): {
+			presenceData.details = "Browsing latest released novels";
+			break;
+		}
+		case pathname.includes("/latest-novel/"): {
+			presenceData.details = "Browsing the latest novels";
+			break;
+		}
+		case pathname.includes("/completed-novel/"): {
+			presenceData.details = "Browsing fully finished novels";
+			break;
+		}
+		case pathname.includes("/most-popular-novel/"): {
+			presenceData.details = "Browsing the most popular novels";
+			break;
+		}
+		case pathname.includes("/genre/"): {
+			presenceData.details = `Checking out ${document
+				.querySelector<HTMLHeadingElement>('[class="tit"]')
+				.textContent.toLowerCase()}`;
+			break;
+		}
+		case pathname.includes("/author/"): {
+			presenceData.details = "Checking out an author";
+			presenceData.state = document.querySelector<HTMLAnchorElement>(
+				"body > div.header > div.cur > div > a:nth-child(5)"
+			).title;
+			if (showButtons) {
+				presenceData.buttons = [
+					{
+						label: "View Author",
+						url: location.href
+					}
+				];
+			}
+			break;
+		}
+		case pathname === "/search/": {
+			presenceData.details = "Searching for a novel";
+			presenceData.state = `"${document.querySelector(
+				"body > div.main > div.wp > div.m-t1 > em.e2"
+			)}`;
```

Comment:
pretty sure stringifying a query selector won't really work. Also you don't have closing quotes here, only opening ones. Feel free to show me a screenshot if I'm wrong though

Reply:
> pretty sure stringifying a query selector won't really work. Also you don't have closing quotes here, only opening ones. Feel free to show me a screenshot if I'm wrong though

Still having a problem debugging, been happening for a while, so bear with me while I try to get this review set